src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/volume/vanilla/BucketFluidVolume.java

```diff
@@ -0,0 +1,14 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.vanilla;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.access.BucketItemAccess;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.ImmutableFluidVolume;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+
+public class BucketFluidVolume extends ImmutableFluidVolume /*unfortunately, due to the nature of items in mc and how potions are implemented, it must be immutable*/ {
+	public BucketFluidVolume(ItemStack buckets) {
+		super(((BucketItemAccess) buckets.getItem()).getFluid(), (buckets.getItem() == Items.BUCKET ? Fraction.ZERO : Fraction.ONE).multiply(buckets.getCount()), buckets.getSubTag("fluid_data"));
```

Comment:
```suggestion
		super(((BucketItemAccess) buckets.getItem()).getFluid(), (buckets.getItem() == Items.BUCKET ? Fraction.ZERO : Fraction.ONE).multiply(buckets.getCount()), buckets.getOrCreateSubTag("fluid_data"));
```

Never assume tags exist, make sure they exist or create an empty dummy.